src/mainloop.rkt

```diff
@@ -300,23 +300,19 @@
                     (list (*context*))))
   (cons domain (apply mk-pcontext pts+exs)))
 
-(define (initialize-alt-table! prog pcontext ctx)
-  (define alt (make-alt prog))
-  (*start-prog* prog)
-  (define table (make-alt-table (*pcontext*) alt ctx))
-
-  ; Add starting alt in every precision
-  (^table^
-   (if (*pareto-mode*)
-       (let ([new-alts (starting-alts alt ctx)])
-         (define-values (errss costs) (atab-eval-altns table new-alts ctx))
-         (atab-add-altns table new-alts errss costs))
-       table))
-
-  (when (flag-set? 'setup 'simplify)
-      (reconstruct! (patch-table-run-simplify (atab-active-alts (^table^))))
-      (finalize-iter!)
-      (^next-alts^ #f)))
+(define (initialize-alt-table! simplified context pcontext)
+  (define initial (make-alt (first simplified)))
+  (define table (make-alt-table pcontext initial context))
+  ;; TODO: Need to include starting-alts stuff here
+  (define alternatives
+    (for/list ([expression (in-list (rest simplified))])
+      (alt
+       expression
+       ;; TODO: What do we put here?
+       (list 'simplify null 'todo-egg-query #f #f)
```

Comment:
Good Q. I think you probably want `find-preprocessing` to return `alt`s instead of expressions. That will also be a good setup for giving preprocessing derivations if we do that later.

Reply:
Ah yes, that seems right. I was hesitant to do that earlier because it felt like a separation of concerns issue, but I had forgotten that we plan to include preprocessing instructions in alts later.